### PR TITLE
Add readiness and liveness probe

### PR DESCRIPTION
**Pull request by CatherineF-dev**
Fixes: [#485](https://github.com/GoogleCloudPlatform/k8s-stackdriver/issues/485)

Tested: 
```
curl -k https://localhost:6443/livez
ok

curl -k https://localhost:6443/readyz
ok

k top pods -n custom-metrics
NAME                                                 CPU(cores)   MEMORY(bytes)   
custom-metrics-stackdriver-adapter-cddcdcf54-vdhlj   3m           32Mi 
```

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
The failed test isn't related to this change. 

The failed one is a flaky test, in checking

File: custom-metrics-stackdriver-adapter/deploy/production/adapter.yaml [hunk 1]
```diff
@@ -78,6 +78,26 @@ spec:
         command:
         - /adapter
         - --use-new-resource-model=false
+        - --secure-port=6443
+        ports:
+        - containerPort: 6443
+          name: https
```
Comment (leonzz) [inline custom-metrics-stackdriver-adapter/deploy/production/adapter.yaml:84]: is this port only for serving the readiness and liveness checks?  if so consider giving it a more descriptive name or just use port number on line 89 and 97 without a name.
If it is also serving the metrics API, then I guess we also need to update the `targetPort: 443` for the Service?

Comment (CatherineF-dev) [reply custom-metrics-stackdriver-adapter/deploy/production/adapter.yaml:84]: Good point! Nice catch!
Updated to `targetPort: 6443` and did an E2E test following https://cloud.google.com/stackdriver/docs/managed-prometheus/hpa#stackdriver-adapter

**Review by leonzz — APPROVED** [review]

**Pull request merged**